test/control_filters/test_gravity_compensation.cpp

```diff
@@ -0,0 +1,113 @@
+{
+  std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+    std::make_shared<control_filters::GravityCompensation<geometry_msgs::msg::WrenchStamped>>();
+
+  // wrong vector size, should fail
+  ASSERT_FALSE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+
+  // fixed wrong vector size
+  node_->set_parameter(rclcpp::Parameter("CoG.pos", std::vector<double>({0.0, 0.0, 0.0})));
+  // all parameters correctly set AND second call to yet unconfigured filter
+  ASSERT_TRUE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+
+  // change a parameter
+  node_->set_parameter(rclcpp::Parameter("CoG.pos", std::vector<double>({0.0, 0.0, 0.2})));
+  // accept second call to configure with valid parameters to already configured filter
+  ASSERT_TRUE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+}
+
+
+TEST_F(GravityCompensationTest, TestGravityCompensationMissingTransform)
+{
+  std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+    std::make_shared<control_filters::GravityCompensation<geometry_msgs::msg::WrenchStamped>>();
+
+  // all parameters correctly set
+  ASSERT_TRUE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+
+  geometry_msgs::msg::WrenchStamped in, out;
+  in.header.frame_id = "world";
+  in.wrench.force.x = 1.0;
+  in.wrench.torque.x = 10.0;
+
+  // should fail due to missing datain frame to sensor frame transform
+  ASSERT_FALSE(filter_->update(in, out));
+}
+
+TEST_F(GravityCompensationTest, TestGravityCompensationComputation)
+{
+  std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+    std::make_shared<control_filters::GravityCompensation<geometry_msgs::msg::WrenchStamped>>();
+
+  double gravity_acc = 9.81;
+  double mass = 5.0;
+
+  ASSERT_TRUE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+
+  geometry_msgs::msg::WrenchStamped in, out;
+  in.header.frame_id = "world";
+  in.wrench.force.x = 1.0;
+  in.wrench.torque.x = 10.0;
+
+  // should pass (transform is identity)
+  ASSERT_TRUE(filter_->update(in, out));
+
+  ASSERT_EQ(out.wrench.force.x, 1.0);
+  ASSERT_EQ(out.wrench.force.y, 0.0);
+  ASSERT_NEAR(out.wrench.force.z, gravity_acc * mass, 0.0001);
+
+  ASSERT_EQ(out.wrench.torque.x, 10.0);
+  ASSERT_EQ(out.wrench.torque.y, 0.0);
+  ASSERT_EQ(out.wrench.torque.z, 0.0);
+
+  out.header.frame_id = "base";
+  // should fail due to missing transform for desired output frame
+  ASSERT_FALSE(filter_->update(in, out));
+
+  // TODO(guihomework) Add a test with real lookups
```

Comment:
please create a follow-up issue, describing precisely what you mean by this comment and link it here instead of the text TODO